pkg/collector/prometheus_collector.go

```diff
@@ -16,6 +16,14 @@ import (
 	"k8s.io/metrics/pkg/apis/custom_metrics"
 )
 
+type noResultError struct {
```

Comment:
nit: Probably makes sense to expose the type as it's returned by the public `GetMetrics` method.